EventListener/CampaignSubscriber.php

```diff
@@ -101,7 +101,7 @@ public function onTriggerCampaignPreSave(TriggerCampaignEvent $event): void
                 $this->getTriggerDialogService()->updateCampaign($triggerCampaign);
             }
             if (isset($changes['variables'])) {
-                $this->getTriggerDialogService()->updateCampaignVariable($triggerCampaign);
+                $this->getTriggerDialogService()->updateCampaignVariable($triggerCampaign, $changes['variables'][1]);
```

Comment:
Why `[1]`?

Reply:
In [1] is always the changed new dataset